File: src/main/java/no/nav/veilarboppfolging/controller/ArenaOppfolgingController.java
package no.nav.veilarboppfolging.controller;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import no.nav.common.types.identer.AktorId;
import no.nav.common.types.identer.Fnr;
import no.nav.veilarboppfolging.controller.response.OppfolgingEnhetMedVeilederResponse;
import no.nav.veilarboppfolging.service.ArenaOppfolgingService;
import no.nav.veilarboppfolging.service.AuthService;
import org.springframework.web.bind.annotation.*;

@Slf4j
@RestController
@RequiredArgsConstructor
@RequestMapping("/api/person")
public class ArenaOppfolgingController {

    private final AuthService authService;

    private final ArenaOppfolgingService arenaOppfolgingService;

    /*
     API used by veilarbmaofs. Contains only the necessary information
     */
    @GetMapping("/{fnr}/oppfolgingsstatus")
    public OppfolgingEnhetMedVeilederResponse getOppfolginsstatus(@PathVariable("fnr") Fnr fnr) {
        authService.sjekkLesetilgangMedFnr(fnr);
        return arenaOppfolgingService.getOppfolginsstatus(fnr);
    }

    @GetMapping("/oppfolgingsenhet")
    public OppfolgingEnhetMedVeilederResponse.Oppfolgingsenhet getOppfolgingsenhet(@RequestParam("aktorId") AktorId aktorId) {
        authService.sjekkLesetilgangMedAktorId(aktorId);
        Fnr fnr = authService.getFnrOrThrow(aktorId);
        return arenaOppfolgingService.getOppfolginsstatus(fnr).getOppfolgingsenhet();
    }
}

File: src/main/kotlin/no/nav/veilarboppfolging/controller/ArenaOppfolgingController.kt
package no.nav.veilarboppfolging.controller

import lombok.RequiredArgsConstructor
import lombok.extern.slf4j.Slf4j
import no.nav.common.types.identer.AktorId
import no.nav.veilarboppfolging.oppfolgingsbruker.arena.ArenaOppfolgingService
import no.nav.veilarboppfolging.oppfolgingsbruker.arena.OppfolgingEnhetMedVeilederResponse
import no.nav.veilarboppfolging.service.AuthService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@Slf4j
@RestController
@RequiredArgsConstructor
@RequestMapping("/api/person")
class ArenaOppfolgingController(
    val authService: AuthService,
    val arenaOppfolgingService: ArenaOppfolgingService
) {

    @GetMapping("/oppfolgingsenhet")
    fun getOppfolgingsenhet(@RequestParam("aktorId") aktorId: AktorId?): OppfolgingEnhetMedVeilederResponse.Oppfolgingsenhet? {
        authService.sjekkLesetilgangMedAktorId(aktorId)
        val fnr = authService.getFnrOrThrow(aktorId)
        return arenaOppfolgingService.hentArenaOppfolgingsEnhet(fnr)
    }
}
